webapp/src/containers/Rhs/projectDetails/header.tsx

```diff
@@ -0,0 +1,123 @@
+}
+
+const Header = ({projectDetails, showAllSubscriptions, handlePagination, setShowAllSubscriptions, handleResetProjectDetails, filter, setFilter, setSubscriptionList}: HeaderProps) => {
+    const {projectName} = projectDetails;
+    const [showProjectConfirmationModal, setShowProjectConfirmationModal] = useState(false);
+
+    // const [showFilter, setShowFilter] = useState(false); // TODO: uncomment when need to toggle filter
+
+    const dispatch = useDispatch();
+    const {makeApiRequestWithCompletionStatus, getApiState, state} = usePluginApi();
+
+    // Opens a confirmation modal to confirm unlinking a project
+    const handleUnlinkProject = () => {
+        setShowProjectConfirmationModal(true);
+    };
+
+    // Handles unlinking a project and fetching the modified project list
+    const handleConfirmUnlinkProject = () => {
+        makeApiRequestWithCompletionStatus(plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName, projectDetails);
+    };
+
+    useApiRequestCompletionState({
+        serviceName: plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName,
+        payload: projectDetails,
+        handleSuccess: () => {
+            dispatch(toggleIsLinkedProjectListChanged(true));
+            handleResetProjectDetails();
+            setShowProjectConfirmationModal(false);
+        },
+    });
+
+    const {isLoading: isUnlinkProjectLoading} = getApiState(plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName, projectDetails);
+
+    return (
+        <>
+            <ConfirmationModal
+                isOpen={showProjectConfirmationModal}
+                onHide={() => setShowProjectConfirmationModal(false)}
+                onConfirm={handleConfirmUnlinkProject}
+                isLoading={isUnlinkProjectLoading}
+                confirmBtnText='Unlink'
+                description={`Are you sure you want to unlink ${projectName}?`}
+                title='Confirm Project Unlink'
+            />
+            <div className='rhs-header-divider'>
+                <div className='d-flex align-item-center margin-bottom-15'>
+                    <BackButton onClick={handleResetProjectDetails}/>
+                    <p className='rhs-title'>{projectName}</p>
+                    <PrimaryButton
+                        text='Unlink'
+                        iconName='fa fa-chain-broken'
+                        extraClass='rhs-project-details-unlink-button'
+                        onClick={handleUnlinkProject}
+                    />
+                </div>
+                <div className='d-flex align-item-center'>
+                    <ToggleSwitch
+                        active={showAllSubscriptions}
+                        onChange={(active) => {
+                            handlePagination(true, false);
+                            setShowAllSubscriptions(active);
+                        }}
+                        label={'Show All Subscriptions'}
```

Comment:
Remember to change this label according to the latest changes suggested by Mike.